### PR TITLE
fix(devise): match sign-in emails case-insensitively

### PR DESCRIPTION
**Pull request by ornsteinfilip**
## Problem

`Folio::User.find_for_authentication` replaces Devise's default lookup so authentication can be scoped by site. Because it calls `find_by(email:)` on the raw Warden params, it skips Devise's `case_insensitive_keys` and `strip_whitespace_keys`.

Sign-in then fails when the submitted email differs only by letter case or surrounding whitespace from the stored value. The Devise email field was also rendered as `type="text"` without `autocapitalize="none"`, so mobile keyboards often capitalize the first character.

Password reset can still succeed (it goes through Devise's filtered `find_or_initialize_with_errors`), after which sign-in fails again with invalid email or password.

## Change

- Run Warden params through `devise_parameter_filter` before the site-scoped lookup.
- Downcase and strip the email used when checking pending invitations.
- Render the Devise email field as `type="email"` with `autocapitalize="none"`.

## Tests

- `find_for_authentication` matches mixed-case and whitespace-padded emails
- Session `#create` with mixed-case / padded email increments `sign_in_count`
- Sign-in email input is `type="email"` with `autocapitalize="none"`

### CODE REVIEW

File: app/controllers/concerns/folio/users/devise_controller_base.rb [hunk 1]
```diff
@@ -61,7 +61,7 @@ def after_sign_in
   end
 
   def email_belongs_to_invited_pending_user?(email)
-    user = Folio::User.find_by(email:)
+    user = Folio::User.find_by(email: email.to_s.strip.downcase)
```
Comment (ornsteinfilip) [inline app/controllers/concerns/folio/users/devise_controller_base.rb]: **[suggestion]** The pending-invitation check now strips and downcases the submitted email, but still does an unscoped `Folio::User.find_by(email:)`. Uniqueness is only per auth site, so the same address can exist on more than one site. `find_by` then returns an arbitrary row: a pending invite on another site can hijack sign-in or password-reset into the invitation flow, and an already-accepted user on another site can hide a pending invite on the current site. Mixed-case input used to miss those rows; after this change it matches them. There is also no test that an invited user can be detected when the submitted email differs only by case or surrounding whitespace.

**Suggestion:** Resolve the user the same way sign-in does — current auth site (or `enabled_site_for_crossdomain_devise`) plus the filtered email, e.g. `find_for_authentication(email:, auth_site_id:)` or `site.auth_users.find_by(email: normalized)`. Add a session/password example that posts a mixed-case or padded invited email and still takes the invitation path.

File: app/models/folio/user.rb [hunk 2]
```diff
@@ -392,7 +392,10 @@ def born_at_required?
 
   private
     # Override of Devise method to scope authentication by zone.
+    # Apply Devise case_insensitive_keys / strip_whitespace_keys here: this
+    # method replaces the default lookup, so Warden does not run the filter.
```
Comment (ornsteinfilip) [inline app/models/folio/user.rb]: **[nit]** The new comment says Warden does not run the filter. Warden never did. Devise’s default `find_for_authentication` delegates to `find_first_by_auth_conditions`, which is what applies `devise_parameter_filter`. This override skips that method, so the filter must be applied here.

**Suggestion:** Say that this override replaces Devise’s default lookup (which filters inside `find_first_by_auth_conditions`), so the filter has to run here.
